app/javascript/components/HelloWorld.js

```diff
@@ -0,0 +1,48 @@
+import React from "react"
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+
+const GET_THINGS_REQUEST = 'GET_THINGS_REQUEST';
+const GET_THINGS_SUCCESS = 'GET_THINGS_SUCCESS';
+
+function getThings() {
+  return dispatch => {
+    dispatch({ type: GET_THINGS_REQUEST
+    });
+    return fetch('v1/things.json')
+      .then(response => response.json())
+      .then(json => dispatch(getThingsSuccess(json)))
+      .catch(error => console.log(error))
+  };
+};
+
+export function getThingsSuccess(json) {
+  return {
+    type: GET_THINGS_SUCCESS,
+    json
+  };
+};
```

Comment:
According to the project requirement, please ensure that you create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting instead of having your action in your component, please move them to a separate file. It is a good practice to always group your redux in a folder.